File: subsequence_sum.java
package day1;

import java.util.ArrayList;

public class subsequence_sum {

	static void fun(ArrayList<Integer> arr, String ans, int i, int sum, int k) {
		if (i == arr.size()) {
			if (sum == k)
				System.out.println(ans + " ");
			return;
		}

		// pick

		fun(arr, ans + arr.get(i) + ",", i + 1, sum + arr.get(i), k);
		// drop

		fun(arr, ans, i + 1, sum, k);

	}

	public static void main(String[] args) {
		ArrayList<Integer> arr = new ArrayList();
		arr.add(1);
		arr.add(2);
		arr.add(1);
		int key = 2;
		String ans;
		fun(arr, "", 0, 0, key);// inp ans index sum,key

	}

}

File: subsequence_sum.cpp
#include<bits/stdc++.h>
using namespace std;

void fun(int i,vector<int> &arr,int k,int sum,vector<int> &ans){
    if(i==arr.size()){
        if(sum==k){
        for(int x:ans)
            cout << x <<" ";
        cout<<endl;
        }
        return ;    
    }
    //pick
    ans.push_back(arr[i]);
    sum+=arr[i];
    fun(i+1,arr,k,sum,ans);
    //not pick
    ans.pop_back();
    sum-=arr[i];
    fun(i+1,arr,k,sum,ans);

}

int main(){
    vector<int>arr={1,2,2,5};
    vector<int> ans={};
    int key=5,i=0;
    fun(0,arr,key,0,ans);
    return 0;
}